components/server/src/github/github-context-parser.ts

```diff
@@ -135,6 +137,17 @@ export class GithubContextParser extends AbstractContextParser implements IConte
 
             if (result.errors) {
                 log.error("Errors executing query.", { query, errors: new TrustedValue(result.errors) });
+                const errorMessages = result.errors.map((e: any) => e.message).join(", ");
+                let errorCode: ErrorCode = ErrorCodes.BAD_REQUEST;
+                switch ((result.errors[0] as any)?.type) {
+                    case "NOT_FOUND":
+                        errorCode = ErrorCodes.NOT_FOUND;
+                        break;
+                    case "FORBIDDEN":
+                        errorCode = ErrorCodes.PERMISSION_DENIED;
+                        break;
+                }
+                throw new ApplicationError(errorCode, errorMessages);
```

Comment:
Can we feature-flag this? I'm not clear on the interface here, but IIRC correctly there was a reason we did not check errors before.

Also, let's move it below the other error check in line 153 ( `if (result.data.repository === null)` ) to not break that helper